nsis/src/nsis/NsisAnalyzer.java

```diff
@@ -75,11 +81,86 @@ public boolean added(Program program, AddressSetView set, TaskMonitor monitor, M
-		Disassembler disassembler = Disassembler.getDisassembler(program, monitor, null);
-		AddressSet modifiedAddrSet = disassembler.disassemble(entriesBlock.getStart(), null);
+		AddressSet modifiedAddrSet = disassembleByteCode(program, entriesBlock, monitor);
+
 		if (modifiedAddrSet.isEmpty()) {
 			return false;
 		}
+
+		MemoryBlock stringsBlock = program.getMemory()
+				.getBlock(NsisConstants.STRINGS_MEMORY_BLOCK_NAME);
+		InstructionIterator instructions = program.getListing().getInstructions(modifiedAddrSet,
+				/* forward direction */ true);
+
+		for (Instruction instr : instructions) {
+			try {
+				resolveStrings(instr, stringsBlock);
+			} catch (MemoryAccessException e) {
+				monitor.setMessage(
+						"Unable to revolve strings at instruction: " + instr.getAddressString(
+								/* display mnemonic */ true, /* pad address if necessary */ true));
+			}
+		}
+
 		return true;
 	}
+
+	/**
+	 * Disassembles the byte code in the specified memory block.
+	 * 
+	 * @param program     to instanciate the disassembler with
+	 * @param memoryBlock to perform the disassembly on
+	 * @param monitor     the TaskMonitor object to monitor the operation
+	 * @return the AddressSet of the disassembled instructions
+	 */
+	private AddressSet disassembleByteCode(Program program, MemoryBlock memoryBlock,
+			TaskMonitor monitor) {
+		Disassembler disassembler = Disassembler.getDisassembler(program, monitor,
+				/* Object to notify */ null);
+		AddressSet entriesAddrSet = new AddressSet(memoryBlock.getStart(), memoryBlock.getEnd());
+		return disassembler.disassemble(entriesAddrSet.getMinAddress(), entriesAddrSet,
+				/* follow flow */ true);
+	}
+
+	/**
+	 * Resolve strings for the specified instruction.
+	 * 
+	 * @param instr        the instruction
+	 * @param stringsBlock the memory block containing the strings
+	 * @throws MemoryAccessException
+	 */
+	private void resolveStrings(Instruction instr, MemoryBlock stringsBlock)
+			throws MemoryAccessException {
+
+		String mnemonic = instr.getMnemonicString();
+		int[] arguments = new int[NsisConstants.NUMBER_OF_PARAMETERS];
+		for (int i = 0; i < NsisConstants.NUMBER_OF_PARAMETERS; i++) {
+			arguments[i] = instr.getInt(Integer.BYTES * (i + 1));
+		}
+
+		switch (mnemonic) {
+		case "MessageBox":
+			addReferenceToString(instr, stringsBlock, arguments[1], 1);
+			break;
+
+		default:
+			break;
+		}
+	}
+
+	/**
+	 * Adds a reference to a string on the specified parameter of an instruction.
+	 * 
+	 * @param instr          the related instruction
+	 * @param stringsBlock   the memory block containing the strings
+	 * @param stringOffset   the offset of the string in the memory block
+	 * @param parameterIndex the index of the parameter to put the reference on
+	 */
+	private void addReferenceToString(Instruction instr, MemoryBlock stringsBlock, int stringOffset,
+			int parameterIndex) {
+		Address parameterAddr = stringsBlock.getStart().add(stringOffset);
```

Comment:
Instead of passing in `MemoryBlock stringsBlock` get the value of `stringsBlock.getStart()` and pass that in. Then pass that address in. Avoids calling function over and over when the answer will always be the same.

That may let you get rid of this function completely.

Reply:
good point, I got rid of the function entirely